lib/logflare/backends.ex

```diff
@@ -383,11 +388,15 @@ defmodule Logflare.Backends do
   end
 
   def lookup(module, %Source{} = source) do
-    {:via, _registry, {registry, via_id}} = via_source(source, module)
+    {:via, registry_mod, {registry, via_id}} = via_result = via_source(source, module)
 
-    case Registry.lookup(registry, via_id) do
-      [{pid, _}] -> {:ok, pid}
-      _ -> {:error, :not_started}
+    if registry_mod == :syn do
```

Comment:
I believe you should be able to do `GenServer.whereis(via_result)` and it will dispatch to the mod accordingly :)